File: Clases 2018/Clases-C#/ClasePractica12_2018 (transacciones con data manager)/C#_TPS_BugTracker/TPS_BugTracker/GUILayer/frmPrincipal.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using TPS_BugTracker.BusinessLayer;
using TPS_BugTracker.BusinessLayer.Entities;
using TPS_BugTracker.GUILayer;
using TPS_BugTracker.GUILayer.Bugs;
using TPS_BugTracker.GUILayer.Login;
using TPS_BugTracker.GUILayer.Usuarios;

namespace TPS_BugTracker
{
    public partial class frmPrincipal : Form
    {
        private static Usuario userLogin = null/* TODO Change to default(_) if this is not a reference type */;

        public frmPrincipal()
        {
            InitializeComponent();
            toolStripStatusLabel1.Text = "";
        }

        private void frmPrincipal_Load(object sender, EventArgs e)
        {
            frmLogin login = new frmLogin(this);
            login.ShowDialog();
        }


        public void actualizarUsuarioLogueado(Usuario user)
        {
            toolStripStatusLabel1.Text = "Usuario: " + user.nombre + " | Perfil: " + user.perfil;
            // Asignamos en userLogin el usuario recibido desde el login y lo dejamos como atributo del formulario principal
            // Podemos simular una sesión de usuario y a partir de este momento desde cualquier ventana podemos obtener
            // el usuario logueado mendiante el método obtenerUsuarioLogin de esta clase.
            userLogin = user;
            Mnu_principal.Visible = true;
        }

        private void ConsultaDeBugsToolStripMenuItem_Click(System.Object sender, System.EventArgs e)
        {
            frmBugs Bugs = new frmBugs();
            Bugs.ShowDialog();
        }

        private void SalirToolStripMenuItem_Click(System.Object sender, System.EventArgs e)
        {
            this.Close();
        }

        private void UsuariosToolStripMenuItem_Click(System.Object sender, System.EventArgs e)
        {
            frmUsuarios usuarios = new frmUsuarios();
            usuarios.ShowDialog();
        }

        public static Usuario obtenerUsuarioLogin()
        {
            return userLogin;
        }


        
    }
}

File: PAV 1-TPI - CASOS DE USO version 1/Resueltos/Usuarios+SeleccionActBugs/App_BugTracker resuelto/App_BugTracker/GUILayer/Login/Frm_principal.vb
Public Class Frm_Principal
    Private userLogin As Usuario = Nothing

    Private Sub Frm_Principal_FormClosing(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles Me.FormClosing
        If MsgBox("Seguro que desea salir de la aplicación?", MsgBoxStyle.Question + MsgBoxStyle.OkCancel, "Aviso") = 1 Then
            End
        Else
            'La propiedad e.Cancel permite cancelar el evento recibido.
            e.Cancel = True
        End If
    End Sub
    Private Sub Frm_Principal_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        'Al cargar el formulario principal cargamos y mostramos el formulario: Frm_login en forma MODAL.
        pnl_info_user.Width = Me.Width
        Mnu_principal.Visible = False
        Frm_login.ShowDialog()
    End Sub

    Public Sub actualizarUsuarioLogueado(ByVal user As Usuario)
        lbl_user.Text = "Usuario: " + user.nombre + " | Perfil: " + user.perfil
        'Asignamos en userLogin el usuario recibido desde el login y lo dejamos como atributo del formulario principal
        'Podemos simular una sesión de usuario y a partir de este momento desde cualquier ventana podemos obtener
        'el usuario logueado mendiante el método obtenerUsuarioLogin de esta clase.
        userLogin = user
        Mnu_principal.Visible = True
    End Sub

    Private Sub ConsultaDeBugsToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ConsultaDeBugsToolStripMenuItem.Click
        Frm_Bugs.ShowDialog()
    End Sub

    Private Sub SalirToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles SalirToolStripMenuItem.Click
        Me.Close()
    End Sub

    Private Sub UsuariosToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles UsuariosToolStripMenuItem.Click
        Frm_Users.ShowDialog()
    End Sub

    Public Function obtenerUsuarioLogin() As Usuario
        Return userLogin
    End Function
End Class